apport/packaging_impl/apt_dpkg.py

```diff
@@ -1334,20 +1357,18 @@ def _check_files_md5(sumfile):
         return mismatches
 
     @staticmethod
-    def _get_primary_mirror_from_apt_sources(apt_sources: str) -> str:
+    def _get_primary_mirror_from_apt_sources(apt_dir: str) -> str:
         """Heuristically determine primary mirror from an apt sources.list."""
-        with open(apt_sources, encoding="utf-8") as f:
-            for line in f:
-                source = aptsources.sourceslist.SourceEntry(line, apt_sources)
-                if source.disabled or source.invalid:
-                    continue
-                if source.type != "deb":
-                    continue
-                return source.uri
+        sources = _load_all_sources(apt_dir)
+        for source in sources:
+            if source.disabled or source.invalid:
+                continue
+            if source.type == "deb":
+                return source.uri or ""
 
         raise SystemError(
-            f"cannot determine default mirror:"
-            f" {apt_sources} does not contain a valid deb line"
+            "cannot determine default mirror:"
```

Comment:
```suggestion
            f"cannot determine default mirror:"
```
Probably a linter will complain about that.

Reply:
Hm, pylint does not complain about that and that string plus f-string concatenation seems to work. So you can ignore that.